crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -300,6 +327,36 @@ impl Schedule {
             system.apply_buffers(world);
         }
     }
+
+    /// Returns an iterator over all systems in this schedule.
+    ///
+    /// Note: this method will return [`ScheduleNotInitialized`] if the
+    /// schedule has never been initialized or run.
+    pub fn systems(
+        &self,
+    ) -> Result<impl Iterator<Item = (NodeId, &BoxedSystem)> + Sized, ScheduleNotInitialized> {
+        if !self.executor_initialized {
+            return Err(ScheduleNotInitialized);
+        }
+
+        let iter = self
+            .executable
+            .system_ids
+            .iter()
+            .zip(&self.executable.systems)
+            .map(|(node_id, system)| (*node_id, system));
+
+        Ok(iter)
+    }
+
+    /// Returns the number of systems in this schedule.
+    pub fn systems_len(&self) -> usize {
+        if !self.executor_initialized {
+            self.graph.systems.len()
+        } else {
+            self.executable.systems.len()
+        }
+    }
```

Comment:
I know it's probably unpopular as there are reasons not to use too long names, but I tend to prefer longer function names instead of a clarifying comment, so I would go with:

`pub fn systems_in_schedule`

Reply:
I am not opposed to long names at all.  My only reluctance with the name `systems_in_schedule()` is from the name alone I'd expect to get a vector of systems back, which `systems()` already does.  I ended up on `systems_len()` because I was doing `systems().len()`, but wanted to reduce the performance overhead.

Other options: `systems_count`, `number_of_systems`, `total_systems`, `count_systems`, `schedule_size` ... oh I kind of like `schedule_size`, but `size` feels like bytes to me.

All that said, I'm happy to rename to `systems_in_schedule()`.

Reply:
`system_len` is fine, but if it cant be that then im partial to `systems_count`.

Reply:
I'll leave this as system_len().

Reply:
Leaving this as `systems_len()` as no strong opinions against the current name appeared.